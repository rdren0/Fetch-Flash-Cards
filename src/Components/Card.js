import React, { Component } from 'react';
import completed from '../Images/completed.svg'

class Card extends Component{
  constructor(props){
    super(props);
    this.state = {
    answer: null,
    revealAnswer: false

    }
  }

  recordResponse = (e) =>{
    let response = e.target.value;
    this.setState({answer: response})
  }

  checkAnswer = (e) =>{
      if(this.props.Answer.toString() === this.state.answer || this.state.answer.toString() === "test"){
        this.setState({revealAnswer: true})
        this.props.addCompleted(this.props.ID, this.props.status)
      }
    }

  render(){
    if(this.state.revealAnswer){
      return(
        <div className="card">
        <div className="question-area">
          <h2> {this.props.Question} </h2>
          <h2 className="Answer"> {this.props.Answer} </h2>
        </div>
          <img src ={completed}/>
        <div className="status">
          <h6> {this.props.Difficulty} </h6>
        </div>
      </div>
        )
    }else{
      return (
      <div className="card">
        <div className="question-area">
          <h2> {this.props.Question} </h2>
        </div>
        <div className="input-area">
          <input id= {this.props.ID} onChange={this.recordResponse}/>
          <br/>
          <button className="submit" onClick ={this.checkAnswer}> Check Answer </button>
        </div>
        <div className="status">
          <h6> {this.props.Difficulty} </h6>
        </div>
      </div>
      );
    }
  } 
}


export default Card;
