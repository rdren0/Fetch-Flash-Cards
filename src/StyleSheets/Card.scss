.card{
  display: grid;
  grid-template-rows: 3fr 2fr 40px;
  padding: 8px;
  min-width: 250px;
  max-width: 250px;
  min-height: 265px;
  max-height: 265px;
  border-radius: 15px;
  margin: 10px;
  border: 2px solid black;
  /*box-shadow: 4px 2px 31px -6px #556F7A;*/
  background-color: #556F7A;
}

.card:hover{
  /*box-shadow: 2px 7px 29px 1px #000000;*/
}

input{
  border-radius: 10px;
  text-align: center;
}

.submit:hover{
  background-color: #23ae87;
}
h2{
  font-weight: 200;
  font-size: 22px;
}

img{
  margin-right: auto;
  margin-left: auto;
  height: 50px;
  widows: 50px;
}

.Answer{
  font-weight: 600;
  color: #23ae87;
}