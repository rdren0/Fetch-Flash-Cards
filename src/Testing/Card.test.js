import React from 'react';
import ReactDOM from 'react-dom';
import Card from '../Components/Card';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let mockCard = {
  "ID": 6,
  "Question": "What function directly follows our fetch URL line?",
  "Answer": ".then()",
  "Difficulty": "Easy"
}

const mockCompletedFn = jest.fn()
const testEvent =  {target: { value: 'test' }} 

describe('Card', () => {
  let wrapper;
  
  beforeEach(() =>{
    wrapper = shallow(
      <Card
        ID={mockCard.ID}
        Question={mockCard.Question}
        Answer={mockCard.Answer}
        Difficulty = {mockCard.Difficulty}
        completed = {false}
        addCompleted = {mockCompletedFn}
        status = {false}

      />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the proper defualt state', () => {
    expect(wrapper.state('answer')).toEqual(null);
    expect(wrapper.state('revealAnswer')).toEqual(false);  
  });


  it('should match the snapshot with all data passed in correctly', () => {
    wrapper.setState({ revealAnswer: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should only show the guess side on uncompleted cards', () =>{
    wrapper = shallow(
        <Card  status={true} />
      )
    expect(wrapper).toMatchSnapshot();
  });

  it('should record a response', () => {
     wrapper.find('input').simulate('change', testEvent)
    expect(wrapper.state("answer")).toEqual("test")
  });

  it('should check if the answer is correct with correct answer', () =>{
    wrapper.setState({ answer: "test" });
    wrapper.find('#submit').simulate('click');
    wrapper.instance().checkAnswer(testEvent)
    expect(wrapper.state("revealAnswer")).toEqual(true);
  });
  it('should check if the answer is correct with wrong answer', () =>{
    wrapper.setState({ answer: ".error()" });
    wrapper.find('#submit').simulate('click');
    wrapper.instance().checkAnswer(".error()")
    expect(wrapper.state("revealAnswer")).toEqual(false);
  });


});


