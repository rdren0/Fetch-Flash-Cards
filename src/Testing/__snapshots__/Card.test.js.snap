// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<div
  className="card"
>
  <div
    className="question-area"
  >
    <h2>
       
      What function directly follows our fetch URL line?
       
    </h2>
  </div>
  <div
    className="input-area"
  >
    <input
      id={6}
      onChange={[Function]}
    />
    <br />
    <button
      className="submit"
      id="submit"
      onClick={[Function]}
    >
       Check Answer 
    </button>
  </div>
  <div
    className="status"
  >
    <h6>
       
      Easy
       
    </h6>
  </div>
</div>
`;

exports[`Card should match the snapshot with all data passed in correctly 2`] = `
<div
  className="card"
>
  <div
    className="question-area"
  >
    <h2>
       
      What function directly follows our fetch URL line?
       
    </h2>
    <h2
      className="Answer"
    >
       
      .then()
       
    </h2>
  </div>
  <img
    alt="Green check mark for completion"
    src="completed.svg"
  />
  <div
    className="status"
  >
    <h6>
       
      Easy
       
    </h6>
  </div>
</div>
`;

exports[`Card should only show the guess side on uncompleted cards 1`] = `
<div
  className="card"
>
  <div
    className="question-area"
  >
    <h2>
       
       
    </h2>
    <h2
      className="Answer"
    >
       
       
    </h2>
  </div>
  <img
    alt="Green check mark for completion"
    src="completed.svg"
  />
  <div
    className="status"
  >
    <h6>
       
       
    </h6>
  </div>
</div>
`;
